packages/nvidia_nat_core/src/nat/profiler/parameter_optimization/parameter_optimizer.py

```
@@ -117,10 +135,62 @@ async def _run_all_evals():
         with (out_dir / f"config_numeric_trial_{trial_id_padded}.yml").open("w") as fh:
             yaml.dump(cfg_trial.model_dump(), fh)
 
-        all_scores = asyncio.run(_run_all_evals())
+        all_results = asyncio.run(_run_all_evals())
+        all_scores = [r[0] for r in all_results]
+        last_eval_output = all_results[-1][1]  # Use last rep for per-item data
         # Persist raw per‑repetition scores so they appear in `trials_dataframe`.
```

Ruff (0.15.0)

[warning] Comment contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)